parsers/MY_WM.py

```diff
@@ -29,65 +36,77 @@
 PRODUCTION_URL = f"https://{DOMAIN}/SystemData/CurrentGen.aspx/GetChartDataSource"
 SOLAR_URL = f"https://{DOMAIN}/SystemData/LargeScaleSolar.aspx/ForecastChart"
 
+PRODUCTION_BREAKDOWN = {
+    "Coal": "coal",
+    "Gas": "gas",
+    "CoGen": "unknown",
+    "Oil": "oil",
+    "Hydro": "hydro",
+    "Solar": "solar",
+}
+
 
 @config.refetch_frequency(timedelta(minutes=10))
 def fetch_consumption(
-    zone_key: str = DEFAULT_ZONE_KEY,
-    session: Optional[Session] = None,
+    zone_key: ZoneKey = DEFAULT_ZONE_KEY,
+    session: Session = Session(),
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> list:
     """Request the power consumption (in MW) of a given zone."""
     date_string = arrow.get(target_datetime).to(TIMEZONE).format("DD/MM/YYYY")
-    return [
+    consumption_data = get_api_data(
+        session or Session(),
+        CONSUMPTION_URL,
         {
-            "datetime": arrow.get(breakdown["DT"], tzinfo=TIMEZONE).datetime,
-            "consumption": breakdown["MW"],
-            "source": DOMAIN,
-            "zoneKey": zone_key,
-        }
-        for breakdown in get_api_data(
-            session or Session(),
-            CONSUMPTION_URL,
-            {
-                "Fromdate": date_string,
-                "Todate": date_string,
-            },
+            "Fromdate": date_string,
+            "Todate": date_string,
+        },
+    )
+    all_consumption_data = TotalConsumptionList(logger)
+
+    for item in consumption_data:
+        all_consumption_data.append(
+            zoneKey=zone_key,
+            datetime=arrow.get(item["DT"]).to(TIMEZONE).datetime,
+            consumption=item["MW"],
+            source=DOMAIN,
         )
```

Comment:
Are we sure this is measured data? The future data points seems to indicate this is forcasted data instead.